### ISSUE TITLE
Simple loop error

### ISSUE THREAD

**Issue by lp5510** [eclarson/PythonIntroBridge#1]
Dr. Larson,

I am working the example below in the Anaconda env using the class Jupyter notebook, and I have encountered the following error which says int_val is not defined. I have checked the code several times and can only conclude that this variable does not exist in the "DataMining" list (?).

I would certainly appreciate your thoughts,

Luke Pierce
   
Slide Type

1 #=======================================================
2 # for loop example with lists
3 print("\nfor loop output:")
4 ​
5 list_example = [int_val, long_val, float_val, bool_val]
6 list_example.insert(0, "DataMining")
7​
8 # notice that the loop ends with a colon and
9 # is designated by the tab alignment
10​
11 for val in list_example:
12     print(str(val) + ' ' + str(type(val)))
13     print("This statement is in the loop")
14        
15 print("This statement is outside the loop")          

for loop output:
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-1-2988ad0501e5> in <module>()
      3 print("\nfor loop output:")
      4 
----> 5 list_example = [int_val, long_val, float_val, bool_val]
      6 list_example.insert(0, "DataMining")
      7 

NameError: name 'int_val' is not defined

**Comment by eclarson** [comment]
I can see the definition of int_val in the code above this. Are you running the Jupiter cells in the order specified?

**Comment by eclarson** [comment]
Just verified this runs properly if you are running all above code in the notebook. Seems to be user error so closing this issue

**Comment by lp5510** [comment]
Got it!
Thanks!

**Comment by lp5510** [comment]
Dr.  Larson,

I am having some difficulty installing rpy2 into my Anaconda notebook, as shown below:
[cid:image002.png@01D32FA1.458C7770]

I have searched this command for syntax changes, but it appears to be correct. I also attempted this command (as some have suggested) in a terminal window to no avail. Any thoughts on how to proceed?

It seems that some of the most trivial errors can be the most challenging.

It would be really helpful if there was a “help desk” arrangement for this course since nothing beats live interaction, and I am sure you have much better things to do than correct basic coding questions from a novice.

Thanks,

Luke

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10
